src/services/config.js

```diff
@@ -119,19 +129,21 @@ class ConfigData {
     }
   }
 
-  addProfile(id, accountSid, region) {
-    // Clean all the inputs.
+  addProfile(id, accountSid, region, apiKey, apiSecret) {
+    //  Clean all the inputs.
     id = this.sanitize(id);
     accountSid = this.sanitize(accountSid);
     region = this.sanitize(region);
 
     const existing = this.getProfileById(id);
+
+    //  Remove if existing in historical projects
     if (existing) {
-      existing.accountSid = accountSid;
-      existing.region = region;
-    } else {
-      this.projects.push(new ConfigDataProfile(id, accountSid, region));
+      this.projects = this.projects.filter((p) => p.id !== existing.id);
```

Comment:
When we are removing the existing project from `this.projects`, don't we need to remove the existing Api keys from keytar?

Reply:
Great point! 
I'll pick this up as part of [DI-1352](https://issues.corp.twilio.com/browse/DI-1352) for profile removal, and marked a comment for updation.